src/+otp/+pendulumdae/PendulumDAEProblem.m

```diff
@@ -0,0 +1,87 @@
+classdef PendulumDAEProblem < otp.Problem
+    %CONSTRAINEDPENDULUM PROBLEM This is a Hessenberg Index-2 DAE problem posed in terms of three constraints
+    %
+    
+    properties (SetAccess=protected)
+        RHSDifferential
+        RHSAlgebraic
+    end
+
+    properties (Dependent)
+      Y0Differential
+      Z0Algebraic
```

Comment:
I'm not opposed to having these, but if we do include this, I think all DAE problems should have these for consistency.  Also, I think `Y0Algebraic` would be better than `Z0Algebraic`.

Reply:
Doesn't really differentiate the two. Algebraic implies there is no direct differential equation thus it is a variable obtained from differentiation of the constraint. I think it should stay.

Reply:
I think this naming is pretty clear.  It's mostly motivated by established naming conventions.  For partitioned problems, we use <property><partition name>.  E.g.

RHSStiff and RHSNonstiff
jacobianDifferential and jacobianAlgebraic
Y0Position and Y0Velocity

This is nice for autocompletion as well.

Reply:
My bad I misinterpreted what was being proposed. I thought you wanted `Y0Differential -> Y0Algebraic` not `Z0Algebraic -> Y0Algebraic`. I agree on the convention you proposed.

Reply:
I was thinking of changing it to that already, glad to see we all think alike